### PR TITLE
[bitnami/grafana-loki] Add existingConfigmap option for Gateway component

### PR DESCRIPTION
**Pull request by tonyjsolano**
### Description of the change

Adds the ability to use an existing configMap for the Gateway component

### Benefits

Able to adjust NGINX configurations without worry of overwriting the next time the HELM chart is deployed

### Possible drawbacks

None

### Applicable issues

- fixes https://github.com/bitnami/charts/issues/18846

### Checklist

- [ X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/grafana-loki/templates/_helpers.tpl [hunk 1]
```diff
@@ -154,6 +154,17 @@ Get the promtail configuration configmap.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Get the NGINX configuration configmap.
+*/}}
+{{- define "grafana-loki.gateway.configmapName" -}}
```
Comment (juan131) [inline bitnami/grafana-loki/templates/_helpers.tpl:160]: You're not using this helper anywhere. I guess you should use it at `bitnami/grafana-loki/templates/gateway/deployment.yaml` updating the section below:
```diff
      volumes:
        - name: config
          configMap:
-           name: {{ include "grafana-loki.gateway.fullname" . }}
+           name: {{ include "grafana-loki.gateway.configmapName" . }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for this contribution! Please check my comments.

File: bitnami/grafana-loki/templates/_helpers.tpl (hunk 1 above)
Comment (tonyjsolano) [reply bitnami/grafana-loki/templates/_helpers.tpl:160]: This is correct.  I knew I forgot something.

**Review by juan131 — APPROVED** [review]
LGTM

**Review by juan131 — CHANGES_REQUESTED** [review]
I just realized you need to introduce another change in `bitnami/grafana-loki/templates/gateway/deployment.yaml` (line 34):
```diff
      annotations:
+       {{- if not .Values.gateway.existingConfigmap }}
        checksum/configmap: {{ include (print $.Template.BasePath "/gateway/configmap-http.yaml") . | sha256sum }}
+       {{- end }}
```

**Pull request closed without merging**